table.go

```diff
@@ -292,21 +136,50 @@ func (rt *RoutingTable) addPeer(p peer.ID) (evicted peer.ID, err error) {
 
 		// push the peer only if the bucket isn't overflowing after slitting
 		if bucket.len() < rt.bucketsize {
-			bucket.pushFront(&PeerInfo{p, PeerStateActive})
+			bucket.pushFront(&PeerInfo{Id: p})
 			rt.PeerAdded(p)
-			return "", nil
+			return true, nil
+		}
+	}
+
+	// the bucket to which the peer belongs is full. Let's try to find a peer
+	// in that bucket with a lastSuccessfulOutboundQuery value above the maximum threshold and replace it.
+	allPeers := bucket.peers()
+	for _, pc := range allPeers {
+		if !pc.lastSuccessfulOutboundQuery.IsZero() &&
```

Comment:
You have the nail on the head. 

We are skipping it intentionally as we DO NOT want to evict peers that have been freshly added and haven't yet got a chance to be useful to us.

However, now that I think about it, a peer that's never been useful to us will stay put in the Routing Table as long as it successfully replied to queries, even with bogus replies. Do we want to evict fresh peers or maybe have a `addedAtTime` field for each peer so we can evict them if their `lastSuccessfulOutboundQuery ` is Zero but they've already been around for too long.

Reply:
Yes, the IsZero isn't really going to work as is. The three options I see are:

1) Ignore the Zero case. This means that peers the get added to the routing table by querying us (or by being connected to us when we drop below the minimum threshold in the DHTs fixLowPeers function) can easily get rotated out unless we start using them for queries.
2) Have a separated `addedAtTime` and relevant `addedAtThreshold`
3) Just set the `lastSuccessfulOutboundQuery` to now when we add them to the routing table. This will give them effectively one round of catching up before we consider dropping them.

@petar I'm partial to option 3 here, WDYT?

Reply:
@aarshkshah1992 @petar WDYT about in `TryAdd` just setting the `lastSuccessfulOutboundQuery` to `time.Now()`. We'll end up overwriting it later if we're actually doing an outbound query, otherwise if it's an inbound query we give it some time to be useful to us during a future bucket refresh/lookup.

Reply:
@aschmahmann Makes complete sense to me. Have made the change.